From ab947c1b30bb61615a83b8d1554dd530cc6e99a2 Mon Sep 17 00:00:00 2001
From: Brendan King <Brendan.King@imgtec.com>
Date: Thu, 17 Oct 2024 17:17:46 +0100
Subject: [PATCH 70/71] egl/drm: Allow surfaces with protected content to have
 modifiers

Allow surfaces with protected content to have modifiers, via the
gbm_bo_create_with_modifiers2 path in get_back_bo.

Upstream-Status: Inappropriate [oe specific]

---
 src/egl/drivers/dri2/platform_drm.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/src/egl/drivers/dri2/platform_drm.c b/src/egl/drivers/dri2/platform_drm.c
index e63392e856b..38cd71f0f1d 100644
--- a/src/egl/drivers/dri2/platform_drm.c
+++ b/src/egl/drivers/dri2/platform_drm.c
@@ -316,23 +316,24 @@ get_back_bo(struct dri2_egl_surface *dri2_surf)
    if (dri2_surf->back == NULL)
       return -1;
    if (dri2_surf->back->bo == NULL) {
-      if (surf->base.v0.modifiers && surf->base.v0.flags)
+      uint32_t flags = surf->base.v0.flags;
+
+      if (dri2_surf->base.ProtectedContent)
+         flags |= GBM_BO_USE_PROTECTED;
+
+      if (surf->base.v0.modifiers && flags)
          dri2_surf->back->bo = gbm_bo_create_with_modifiers2(
             &dri2_dpy->gbm_dri->base, surf->base.v0.width, surf->base.v0.height,
             surf->base.v0.format, surf->base.v0.modifiers, surf->base.v0.count,
-            surf->base.v0.flags);
+            flags);
       else if (surf->base.v0.modifiers)
          dri2_surf->back->bo = gbm_bo_create_with_modifiers(
             &dri2_dpy->gbm_dri->base, surf->base.v0.width, surf->base.v0.height,
             surf->base.v0.format, surf->base.v0.modifiers, surf->base.v0.count);
-      else {
-         unsigned flags = surf->base.v0.flags;
-         if (dri2_surf->base.ProtectedContent)
-            flags |= GBM_BO_USE_PROTECTED;
+      else
          dri2_surf->back->bo =
             gbm_bo_create(&dri2_dpy->gbm_dri->base, surf->base.v0.width,
                           surf->base.v0.height, surf->base.v0.format, flags);
-      }
    }
    if (dri2_surf->back->bo == NULL)
       return -1;
-- 
2.34.1

